### PR TITLE
Add more coverage for the Parser class.

### PR DESCRIPTION
**Pull request by Weysan**

### CODE REVIEW

File: test/ParserTest.php [hunk 1]
```diff
@@ -5,15 +5,92 @@
...
-    public function testParsesPartialResponseCorrectly() {
-        $parser = new Parser(function ($result) use (&$parsed) {
-            $parsed = $result;
+class ParserTest extends TestCase
+{
+    /**
+     * @param mixed &$parsed_elements
+     * @return Parser
+     */
+    protected function getParserToTest(&$parsed_elements)
```
Comment (kelunik) [inline test/ParserTest.php]: Please use `$parsedElements`.

File: test/ParserTest.php [hunk 2]
```diff
@@ -5,15 +5,92 @@
...
+    {
+        return new Parser(function ($result) use (&$parsed_elements) {
+            $parsed_elements = $result;
         });
+    }
+
+    public function testParsesPartialResponseCorrectly()
+    {
+        $parser = $this->getParserToTest($parsed);
```
Comment (kelunik) [inline test/ParserTest.php]: You can just create a property and define a `setUp()` method that creates a new parser.

**Comment by Weysan** [comment]
I will make the changes and commit them this week end @kelunik thanks for your feedback!

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**